concordium-std-derive/src/lib.rs

```diff
@@ -1939,6 +1943,113 @@ fn concordium_test_worker(_attr: TokenStream, item: TokenStream) -> syn::Result<
+
+#[cfg(feature = "concordium-quickcheck")]
+#[proc_macro_attribute]
+/// Derive the appropriate export for an annotated QuickCheck function by
+/// exposing it as `#[concordium_test]`. The macro was copied from `QuickCheck`
+/// and changed to use `concordium_std::test_infrastructure::concordium_qc`
+/// instead of the standard  `QuickCheck`'s `quickcheck`
+///
+/// The macro takes a `tests` attrubute that specifies how many tests to run:
+/// `#[concordium_quickcheck(tests = 1000)]`. If no `tests` is provided,
+/// `DEFAULT_QUICKCHECK_TESTS` is used
+pub fn concordium_quickcheck(attr: TokenStream, input: TokenStream) -> TokenStream {
+    use syn::AttributeArgs;
+
+    let output = match syn::Item::parse.parse(input.clone()) {
+        Ok(syn::Item::Fn(mut item_fn)) => {
+            // Parse attrubutes and fail early if wrong atributes are specified
+            let a = attr.clone();
+            let parsed_attr = parse_macro_input!(a as AttributeArgs);
+            let tests = if parsed_attr.is_empty() {
+                DEFAULT_QUICKCHECK_TESTS
+            } else if parsed_attr.len() == 1 {
+                let meta = parsed_attr.get(0).unwrap();
+                match get_quickcheck_tests_count(meta).map_err(|e| e.to_compile_error()) {
+                    Ok(v) => v,
+                    Err(e) => return e.into(),
+                }
+            } else {
+                return syn::parse::Error::new(
+                    proc_macro2::TokenStream::from(attr).span(),
+                    "expected a single `tests = <number>` attribute",
+                )
+                .to_compile_error()
+                .into();
+            };
+            let mut inputs: Punctuated<syn::BareFnArg, syn::token::Comma> =
+                syn::punctuated::Punctuated::new();
+            let mut errors = Vec::new();
+
+            item_fn.sig.inputs.iter().for_each(|input| match *input {
+                syn::FnArg::Typed(syn::PatType {
+                    ref ty,
+                    ..
+                }) => {
+                    inputs.push(parse_quote!(_: #ty));
+                }
+                _ => errors.push(syn::parse::Error::new(
+                    input.span(),
+                    "unsupported kind of function argument",
+                )),
+            });
+
+            if errors.is_empty() {
+                let attrs = mem::take(&mut item_fn.attrs);
+                let name = &item_fn.sig.ident;
+                let codomain = &item_fn.sig.output;
+                quote! {
+                    #[concordium_test]
+                    #(#attrs)*
+                    fn #name() {
+                        #item_fn
+                       ::concordium_std::test_infrastructure::concordium_qc(#tests, #name as (fn (#inputs) #codomain))
+                    }
+                }
+            } else {
+                errors.iter().map(syn::parse::Error::to_compile_error).collect()
+            }
+        }
+        _ => {
```

Comment:
I think it might make sense to split this case into two, `Ok(_)` with the error you already have below, and then `Err(err)` providing the user with the error of why it failed to parse the tokenstream.

Reply:
Done.